cmd/kube-apiserver/app/server.go

```diff
@@ -159,13 +159,13 @@ func CreateServerChain(completedOptions completedServerRunOptions, stopCh <-chan
 		return nil, err
 	}
 
-	kubeAPIServerConfig, sharedInformers, versionedInformers, insecureServingOptions, serviceResolver, pluginInitializer, admissionPostStartHook, err := CreateKubeAPIServerConfig(completedOptions, nodeTunneler, proxyTransport)
+	kubeAPIServerConfig, insecureServingOptions, serviceResolver, pluginInitializer, admissionPostStartHook, err := CreateKubeAPIServerConfig(completedOptions, nodeTunneler, proxyTransport)
 	if err != nil {
 		return nil, err
 	}
 
 	// If additional API servers are added, they should be gated.
-	apiExtensionsConfig, err := createAPIExtensionsConfig(*kubeAPIServerConfig.GenericConfig, versionedInformers, pluginInitializer, completedOptions.ServerRunOptions, completedOptions.MasterCount)
+	apiExtensionsConfig, err := createAPIExtensionsConfig(*kubeAPIServerConfig.GenericConfig, kubeAPIServerConfig.VersionedInformers, pluginInitializer, completedOptions.ServerRunOptions, completedOptions.MasterCount)
```

Comment:
What if we updated `createAPIExtensionsConfig` to take the master.Config to avoid more of these?

Reply:
Yes, anything to get the parameter counts down and make this look more like a logical set up progression. However, I'd prefer that in a followup, as this looks pretty good as-is.

Reply:
@deads2k  @lavalamp not really sure. 😅but it looks odd to me to take sth under master pkg to create an apiextension config. i thought that apiextension is an isolated, inner layer under master and the former one should not import or reference anything from the latter. amiright?